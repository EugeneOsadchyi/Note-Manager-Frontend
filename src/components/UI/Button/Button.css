.Button {
  background-color: transparent;
  border: 2px solid black;
  padding: 7px 12px;
  min-width: 75px;
  margin-right: 5px;
}

.Button:disabled {
  cursor: not-allowed;
  border-color: gray;
  color: gray;
}

.Success {
  font-weight: bold;
}
