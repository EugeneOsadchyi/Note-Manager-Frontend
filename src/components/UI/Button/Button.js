import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Button.css';

const Button = ({
  children, btnType, disabled, clicked,
}) => {
  const classes = classNames('Button', btnType);

  return (
    <button
      type="button"
      disabled={disabled}
      className={classes}
      onClick={clicked}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  btnType: PropTypes.string,
  clicked: PropTypes.func.isRequired,
};

Button.defaultProps = {
  disabled: false,
  btnType: null,
};

export default Button;
