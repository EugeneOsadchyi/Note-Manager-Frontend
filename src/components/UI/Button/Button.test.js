import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Button from './Button';

describe('Button', () => {
  const mockClicked = jest.fn();

  const props = {
    disabled: false,
    btnType: 'Success',
    clicked: mockClicked,
  };

  const children = 'My Button';

  let button = shallow((
    <Button {...props}>
      {children}
    </Button>
  ));

  afterEach(() => mockClicked.mockReset());

  it('renders properly', () => {
    expect(shallowToJson(button)).toMatchSnapshot();
  });

  describe('when button is enabled', () => {
    beforeEach(() => {
      props.disabled = false;

      button = mount((
        <Button {...props}>
          {children}
        </Button>
      ));
    });


    describe('and button was clicked', () => {
      beforeEach(() => button.simulate('click'));

      it('dispatches `clicked()` method it receives from props', () => {
        expect(mockClicked).toHaveBeenCalled();
      });
    });
  });

  describe('when button is disabled', () => {
    beforeEach(() => {
      props.disabled = true;

      button = mount((
        <Button {...props}>
          {children}
        </Button>
      ));
    });

    it('renders properly', () => {
      expect(shallowToJson(button)).toMatchSnapshot();
    });

    describe('and button was clicked', () => {
      beforeEach(() => button.simulate('click'));

      it('does not dispatch `clicked()` method it receives from props', () => {
        expect(mockClicked).not.toHaveBeenCalled();
      });
    });
  });
});
