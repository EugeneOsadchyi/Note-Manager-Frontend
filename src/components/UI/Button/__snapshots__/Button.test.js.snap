// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<button
  className="Button Success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  My Button
</button>
`;

exports[`Button when button is disabled renders properly 1`] = `
<Button
  btnType="Success"
  clicked={[Function]}
  disabled={true}
>
  <button
    className="Button Success"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    My Button
  </button>
</Button>
`;
